app/browse-posts/index.tsx

```diff
@@ -0,0 +1,290 @@
+                    );
+                    const uniqueNewPosts = fetchedPosts.filter(
+                        (post) => !existingIds.has(post.id),
+                    );
+                    return [...prevPosts, ...uniqueNewPosts];
+                });
+
+                setLastPostDoc(docs[docs.length - 1]);
+
+                await fetchItemsForPosts(fetchedPosts.map((post) => post.id));
+            }
+        } catch (error) {
+            console.error(error);
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    async function fetchItemsForPosts(postIds: string[]) {
+        try {
+            const postsToFetch = postIds.filter((id) => !itemsByPost[id]);
+
+            if (postsToFetch.length === 0) return;
+
+            const newItemsByPost: Record<string, ItemType[]> = {};
+
+            for (const postId of postsToFetch) {
+                newItemsByPost[postId] = [];
+            }
+
+            const itemsQuery = query(
+                collection(db, "items"),
+                where("isDeleted", "==", false),
+            );
+
+            const unsubscribeItems = onSnapshot(
+                itemsQuery,
+                (itemsQuerySnapshot) => {
+                    const newItems: ItemType[] = [];
+                    itemsQuerySnapshot.forEach((itemDoc) => {
+                        const data = {
+                            id: itemDoc.id,
+                            ...itemDoc.data(),
+                        } as ItemType;
+                        newItems.push(data);
+                    });
+
+                    for (const postId of postsToFetch) {
+                        newItemsByPost[postId] = newItems.filter(
+                            (item) => item.postId === postId,
+                        );
+                    }
+
+                    setItemsByPost((prev) => ({ ...prev, ...newItemsByPost }));
+                },
+                (error) => {
+                    console.error(error);
+                },
+            );
+
+            setItemUnsubscribers((prev) => [...prev, unsubscribeItems]);
+        } catch (error) {
+            console.error(error);
+        }
+    }
+
+    useEffect(() => {
+        const usersQuery = query(collection(db, "users"));
+        const unsubscribeUsers = onSnapshot(
+            usersQuery,
+            (usersQuerySnapshot) => {
+                const fetchedUsers: UserDataType[] = [];
+                usersQuerySnapshot.forEach((userDoc) => {
+                    const data = { id: userDoc.id, ...userDoc.data() };
+                    fetchedUsers.push(data as UserDataType);
+                });
+                setUsers(fetchedUsers);
+            },
+            (error) => {
+                console.error(error);
+            },
+        );
+
+        return () => unsubscribeUsers();
+    }, []);
+
+    function getUserById(userId: string) {
+        return users.find((user) => user.id === userId);
+    }
```

Comment:
delete, ill create a pr to include the data of the seller in the post, sorry overlooked this 😞